src/main/java/io/github/artsok/extension/ParameterizedRepeatedExtension.java

```diff
@@ -133,7 +133,25 @@ public void handleTestExecutionException(ExtensionContext context, Throwable thr
             throw throwable;
         }
         repeatableExceptionAppeared = true;
-        throw new TestAbortedException("Do not fail completely, but repeat the test", throwable);
+
+        long currentSuccessCount = historyExceptionAppear.stream().filter(exceptionAppeared -> !exceptionAppeared).count();
+        if (currentSuccessCount < minSuccess) {
+            if (isMinSuccessTargetStillReachable(minSuccess)) {
+                throw new TestAbortedException("Do not fail completely, but repeat the test", throwable);
+            } else {
+                throw throwable;
+            }
+        }
+    }
+
+    /**
+     * If cannot reach a minimum success target, will return true
+     *
+     * @param minSuccessCount - minimum success count
+     * @return true/false
+     */
+    private boolean isMinSuccessTargetStillReachable(final long minSuccessCount) {
+        return historyExceptionAppear.stream().filter(bool -> bool).count() <= totalRepeats - minSuccessCount;
```

Comment:
This will make sure that we abort only until we have a success or run out of repetitions. Code is pretty much the same as for the non-parameterized tests.